### ISSUE TITLE
the dev server paused while save some code

### ISSUE THREAD

**Issue by SilentTiger** [webpack/webpack-dev-server#2554]
- Operating System: macos 10.15.4
- Node Version: 12.16.1
- NPM Version: 6.13.4
- webpack Version: 4.43.0
- webpack-dev-server Version: 3.7.2
- Browser: Chrome 81.0.4044.129

- [x] This is a **bug**
- [ ] This is a **modification** request

### Code

```js
// webpack.config.js
const path = require('path')
const os = require('os')
const HtmlWebpackPlugin = require('html-webpack-plugin')
const { CleanWebpackPlugin } = require('clean-webpack-plugin')
const CopyPlugin = require('copy-webpack-plugin')
const HardSourceWebpackPlugin = require('hard-source-webpack-plugin')
const TerserPlugin = require('terser-webpack-plugin')
const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer')

const buildStart = (new Date()).toLocaleString()
console.log(`run on ${os.cpus().length} CPUs`)

const webpackConfig = {
  entry: {
    app: './src/scripts/index.ts',
    normalize: 'normalize.css',
    style: './src/styles/style.scss',
  },
  output: {
    filename: '[name].bundle.js',
    path: path.resolve(__dirname, 'dist'),
    libraryTarget: 'umd',
  },
  devtool: 'source-map',
  devServer: {
    contentBase: path.resolve(__dirname, 'src/assets'),
  },
  mode: 'development',
  plugins: [
    new CleanWebpackPlugin(),
    new HtmlWebpackPlugin({
      template: 'src/assets/template.ejs',
      buildTime: buildStart,
      env: process.env.NODE_ENV,
    }),
    new CopyPlugin([{ from: 'src/assets/sample_docs', to: '../dist/sample_docs' }]),
  ],
  resolve: {
    alias: {
      vue: 'vue/dist/vue.esm.js',
    },
    extensions: ['.tsx', '.ts', '.js'],
  },
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        use: [
          {
            loader: 'ts-loader',
            options: {
              // transpileOnly: true,
              experimentalWatchApi: true,
            },
          },
          {
            loader: 'eslint-loader',
            options: {
              cache: true,
              quiet: true,
            },
          },
        ],
        exclude: /node_modules/,
      },
      {
        test: /\.scss$/,
        use: [
          'style-loader',
          'css-loader',
          'postcss-loader',
        ],
      },
      {
        test: /\.css$/,
        use: [
          'style-loader',
          'css-loader',
        ],
      },
    ],
  },
}

if (process.env.NODE_ENV === 'demo') {
  webpackConfig.optimization = {
    minimize: true,
    minimizer: [new TerserPlugin({
      sourceMap: false,
      parallel: os.cpus().length,
    })],
  }
  webpackConfig.mode = 'production'
  delete webpackConfig.devtool
}

module.exports = webpackConfig

```

```js
// additional code, remove if not needed.
```

### Expected Behavior
Webpack can compile the code after i save some code file.
### Actual Behavior
The webpack start compile the code and paused, it will not continue until i save some code file another time.

### For Bugs; How can we reproduce the behavior?
I found this issue in my project: https://github.com/SilentTiger/pit
this is a new issue of 2020, I'm very sure that in 2019, everything is OK.
<img width="944" alt="Xnip2020-04-30_17-10-40" src="https://user-images.githubusercontent.com/764862/80693651-eddcf680-8b05-11ea-8bb6-7df5be0a98b1.png">

**Comment by SilentTiger** [comment]
I've update all the deps to the latest version, I have tried to remove the cache folder, even remove the hard-source-webpack-plugin, I think I've done all the things I can do.
And another strange thing is, some times, when I save some code and the dev server refresh the page, I found that the source code shown (via source map) in Chrome dev tools has been update, but in fact, Chrome still run the old code.

All the problems comes in these two or three month, everything is allright in 2019, I'm very sure about that.
